collections/mstilkerich/bind9.md

```diff
@@ -0,0 +1,25 @@
+## Bind9 collection
+
+A collection for bind9
+ * Log parser for supporting both logs in syslog and separate bind9 logfile
+ * Scenario that detects bind9 security policy violations
+
+This collection should address the same events as the fail2ban named-refused
+jail.
+
+
+## Acquisition template
+
+Acquisition configuration depends on whether bind9 is configured to log to
+syslog, separate log files, or both.
+
+For a separate log file, set the log type to `named`:
+
+```yaml
+filenames:
+  - /var/log/named/security.log
+labels:
+  type: named
```

Comment:
If type is set to syslog is the program name set the `named` within syslog?

Reply:
Yes, the bind daemon binary is called `named` and uses this also as the tag in syslog messages. Therefore, when log messages are taken from syslog the `syslog-logs` parser will set `Parsed.program` to `named`. Therefore I also use it as label for the bind9 logs, which the `non-syslog` parser will use for setting `Parsed.program` and subsequently the `bind9-logs` parser will see compatible events independent of whether they originate from syslog or a bind9 log file.